go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -511,8 +511,11 @@ func runOvnKube(ctx context.Context, runMode *ovnkubeRunMode, ovnClientset *util
 			// record delay until ready
 			metrics.MetricOVNKubeControllerReadyDuration.Set(time.Since(startTime).Seconds())
 		}()
+		// make sure ovnkubeController started in a separate goroutine will execute .Stop() on shutdown.
+		// Stop() only makes sense to call if Start() succeeded.
 		defer func() {
-			if ovnkubeControllerStartErr != nil {
+			ovnkubeControllerWG.Wait()
+			if ovnkubeControllerStartErr == nil {
```

Comment:
Wouldn't it make sense to get rid of this defer block and just move this down later on the block where we are already checking for the wait group so that it ends up being
```
// wait for ovnkubeController to start and check error
if runMode.ovnkubeController {
    ovnkubeControllerWG.Wait()
    if ovnkubeControllerStartErr != nil {
	return ovnkubeControllerStartErr
    }
    defer cm.Stop()
}
```

Also cluster manager and ovnkube controller use the same variable name `cm` which might get confusing. How would you feel about renaming them?